quickwit/quickwit-ui/src/components/QueryEditor/AggregationEditor.tsx

```diff
@@ -0,0 +1,305 @@
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import { useRef, useState } from 'react';
+import { SearchComponentProps } from '../../utils/SearchComponentProps';
+import { TermAgg, HistogramAgg } from '../../utils/models';
+import { Box } from '@mui/material';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import TextField from '@mui/material/TextField';
+
+export function AggregationEditor(props: SearchComponentProps) {
+  return (
+    <Box hidden={!props.searchRequest.aggregation}>
+      <MetricKind
+        searchRequest={props.searchRequest}
+        onSearchRequestUpdate={props.onSearchRequestUpdate}
+        runSearch={props.runSearch}
+        index={props.index}
+        queryRunning={props.queryRunning} />
+      <AggregationKind
+        searchRequest={props.searchRequest}
+        onSearchRequestUpdate={props.onSearchRequestUpdate}
+        runSearch={props.runSearch}
+        index={props.index}
+        queryRunning={props.queryRunning} />
+    </Box>
+  )
+}
+
+export function MetricKind(props: SearchComponentProps) {
+  // TODO add percentiles
+  const metricRef = useRef(props.searchRequest.aggregationConfig.metric);
+
+  const handleTypeChange = (event: SelectChangeEvent) => {
+    const value = event.target.value;
+    const updatedMetric = value != "count" ? Object.assign({}, metricRef.current, {type: value}) : null;
+    const updatedAggregation = Object.assign({}, props.searchRequest.aggregationConfig, {metric: updatedMetric});
+    const updatedSearchRequest = Object.assign({}, props.searchRequest, {aggregationConfig: updatedAggregation});
+    props.onSearchRequestUpdate(updatedSearchRequest);
+    metricRef.current = updatedMetric;
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    if (metricRef.current == null) {
+      return;
+    }
+    const updatedMetric = Object.assign({}, metricRef.current, {field: value});
+    const updatedAggregation = Object.assign({}, props.searchRequest.aggregationConfig, {metric: updatedMetric});
+    const updatedSearchRequest = Object.assign({}, props.searchRequest, {aggregationConfig: updatedAggregation});
+    props.onSearchRequestUpdate(updatedSearchRequest);
+    metricRef.current = updatedMetric;
+  };
+
+  return (
+    <Box>
+      <FormControl variant="standard" sx={{ m: 1, minWidth: 120, display: 'flex', flexDirection: 'row', }}>
```

Comment:
FormControl is meant to hold only one input component. Multiple inputs will throw errors in console.

Reply:
oh, I saw some warning regarding FormControl, but couldn't figure out their meaning. Thanks!